lib/manager/composer/artifacts.ts

```diff
@@ -73,6 +73,30 @@ function getAuthJson(): string | null {
   return is.emptyObject(authJson) ? null : JSON.stringify(authJson);
 }
 
+function getComposerArguments(config): string {
+  let args = '';
+  if (config.composerIgnorePlatformReqs) {
+    if (config.composerIgnorePlatformReqs.length === 0) {
+      args += ' --ignore-platform-reqs';
+    } else {
+      config.composerIgnorePlatformReqs.forEach((req) => {
+        args += ' --ignore-platform-req ' + quote(req);
+      });
+    }
+  }
+
+  args += ' --no-ansi --no-interaction';
+  if (!getGlobalConfig().allowScripts || config.ignoreScripts) {
+    args += ' --no-scripts --no-autoloader';
+  }
+
+  if (!getGlobalConfig().allowPlugins || config.ignorePlugins) {
```

Comment:
```suggestion
  if (getGlobalConfig().allowPlugins && !config.ignorePlugins) {
```

bit easier to understand

Reply:
I misread this yesterday, but unfortunately this changes the behaviour as we need to disable the plugins when not allowed, instead of the other way around. This suggestion would need to be wrapped with a negation, making it less understandable. This notation was already used by the way: https://github.com/renovatebot/renovate/pull/11990/files#diff-d5d778f4c4a0495c698f14938cd1deb6da79f83e62ff015909a705ef1a0f9319L142